### ISSUE TITLE
jRuby 9.0.1.0 cannot encode Chinese characters with GB18030

### ISSUE THREAD

**Issue by southwolf** [jruby/jruby#3417]
```ruby

puts ENV_JAVA['file.encoding']  # => "UTF-8"
str = "Github 同性交友排解寂寞. jRuby + Java 起夜急开发"
str.encoding           # => #<Encoding:UTF-8> 
str.encode("GB2312")   # => OK
str.encode("GBK")      # => OK, same as GB2312
str.encode("GB18030")  # => should be same as GBK/GB2312
# MRI: OK
# jRuby 9.0.1.0, Oracle JDK 1.8.0_60, Ubuntu 15.04:
# got Encoding::InvalidByteSequenceError: "\xE5\x90\x8C" followed by "\xE6" on UTF-8
```

**Comment by southwolf** [comment]
Might be duplicate with #3411

**Comment by southwolf** [comment]
Already fixed in jruby 9.0.3.0
